doc-content/enterprise-only/openshift/customimage-bc-githooks-proc.adoc

```diff
@@ -0,0 +1,46 @@
+[id='customimage-bc-githooks-proc_{context}']
+:componentimage: businesscentral
+:component: {CENTRAL}
+= Creating a custom {CENTRAL} image with Git hooks
+
+You can create a custom {CENTRAL} image that embeds Git hooks. In an authoring environment, you can use Git hooks to facilitate interaction between the internal Git repository of {CENTRAL} and an external Git repository. 
+
+[NOTE]
+====
+You can also provide Git hooks using a persistent volume or configuration map. For instructions about this method of providing Git hooks, see <<operator-deploy-central-proc>>.
+====
+
+.Procedure
+
+. Prepare the necessary Git hooks script files. This example shows one script file names `pre-commit`. You cam use as many files as necessary. Ensure that the execute permission is set for all script files.
+. If necessary, prepare an `.ssh` directory to enable access to the Git repository using SSH keys. 
+include::customimage-download-step.adoc[]
+. Create a `Dockerfile` that defines a custom image based on the base image. The file must copy the script files into the `/opt/kie/data/git/hooks/` directory. If you prepared an `.ssh` directory, the file must copy it into the `/home/jboss` directory. The following example shows the content of the `Dockerfile` file:
++
+[subs="attributes,verbatim,macros,quotes"]
+----
+FROM registry.redhat.io/{PRODUCT_INIT}-7/{PRODUCT_INIT}-businesscentral-rhel8:7.7.0
+COPY pre-commit /opt/kie/data/git/hooks/
+COPY .ssh /home/jboss/.ssh/
```

Comment:
I am not confident this is the right way to go about this, even though it is the simple example I gave you in the jira. I would like to confirm with @ruromero first.

Ruben, in the email you had forwarded me (which I believe had Misha in the thread for originally), the example for adding ssh keys for git hooks looked like this:

```
SSH Keys
Create secret from file
$ oc create secret generic --from-file=id_rsa=mycert rhpam-centr-gitkey
$ oc set volume dc/rhpam-trial-rhpamcentr --add --type secret --secret-name rhpam-centr-gitkey --mount-path=/home/jboss/.ssh --name=ssh-key

GIT HOOKS (thru configmap) - for small scripts
$ oc create configmap git-hooks --from-file=post-commit=mypostcommit --from-file=pre-commit=myprecommit
$ oc set volume dc/rhpam-trial-rhpamcentr --add --type configmap --configmap-name git-hooks --mount-path=<githooks_dir> --name=git-hooks
```

I'm worried the simple Dockerfile example in this PR is either too simplistic, or might clobber (or be clobbered by) other things. How might you suggest this example be handled in the best way (and if necessary, how should the Dockerfile be changed)? For easier reading, the rendered doc is here:
http://file.ork.redhat.com/~mramendi/BXMSDOC-5891/#customimage-bc-githooks-proc_openshift-operator

Thank you!

Reply:
In my understanding we use *either* config maps *or* custom images for Git Hooks. Moreover, as far as I understand, with the operator we can't use config maps for the SSH directory the way it is described in this example, simply because things would be immediately "restored" by the operator. And there does not seem to be a place in the Console UI to provide SSH directory configmap or secret.

Reply:
@errantepiphany @mramendi this is the way I'd extend the base image (or any other custom image) to include the ssh keys and githooks. What do you think the possible side effects could be?
Maybe it lacks mentioning that this process might need to be adapted if the image used differs from the base image. i.e. custom JDK but IMO if a user is experienced enough to extend the base image he/she should be able to understand how this process needs to be adapted.

Reply:
@ruromero , @mramendi : Okay, I am fine with that. I just wanted to make sure since I am not a subject matter expert (at all) in this area. Thanks!